oci/spec_opts.go

```diff
@@ -1262,15 +1262,15 @@ var ErrNoShmMount = errors.New("no /dev/shm mount specified")
 // The size value is specified in kb, kilobytes.
 func WithDevShmSize(kb int64) SpecOpts {
 	return func(ctx context.Context, _ Client, c *containers.Container, s *Spec) error {
-		for _, m := range s.Mounts {
-			if m.Source == "shm" && m.Type == "tmpfs" {
-				for i, o := range m.Options {
-					if strings.HasPrefix(o, "size=") {
-						m.Options[i] = fmt.Sprintf("size=%dk", kb)
-						return nil
+		for mi, m := range s.Mounts {
+			if m.Destination == "/dev/shm" && m.Source == "shm" && m.Type == "tmpfs" {
+				for oi := 0; oi < len(m.Options); oi++ {
+					if strings.HasPrefix(m.Options[oi], "size=") {
+						m.Options = append(m.Options[:oi], m.Options[oi+1:]...)
+						oi--
 					}
 				}
-				m.Options = append(m.Options, fmt.Sprintf("size=%dk", kb))
+				s.Mounts[mi].Options = append(m.Options, fmt.Sprintf("size=%dk", kb))
```

Comment:
would you mind to add the unit test for this? Thanks

Reply:
okay